### ISSUE TITLE
prevent manual text input

### ISSUE THREAD

**Issue by andytwoods** [jifalops/datetime_picker_formfield#7]
Hi,
just wondering if it would be best to prevent people typing in dates by hand? Or having a little logic to disable this if wanted. The former is achievable by removing parent.controller.text.isEmpty && previousValue.isEmpty. Want a pull request?

cheers,
Andy.

```
void inputChanged() {
    if (//parent.controller.text.isEmpty &&
        //_previousValue.isEmpty &&
       parent.focusNode.hasFocus) {
      getDateTimeInput(context).then((date) {
        parent.focusNode.unfocus();
        setState(() {
          parent.controller.text = _toString(date, parent.format);
          setValue(date);
        });
      });
    } else if (parent.resetIcon != null &&
        parent.controller.text.isEmpty == showResetIcon) {
      setState(() => showResetIcon = !showResetIcon);
      // parent.focusNode.unfocus();
    }
    _previousValue = parent.controller.text;
    if (!parent.focusNode.hasFocus) {
      setValue(_toDate(parent.controller.text, parent.format));
    }
  }
```

**Comment by jifalops** [comment]
I'm on mobile so I can't check right now, but I have a strong suspicion removing those lines will have side effects. I'll look into adding an editable option.

**Comment by jifalops** [comment]
So I started implementing this but having a text field that is focusable but not editable isn't supported in the platform. The best option I could see was to override any changes to the text that were made manually, but then I realized instead of putting in an ugly hack, your use case shouldn't actually use this widget. 

You should just use a `Text` or similar non-editable widget and use the platform's `showDatePicker()` and/or `showTimePicker()` when a user clicks on it.

**Comment by andytwoods** [comment]
Perhaps an ugly way to do it is below. 

```
  FocusNode myFocusNode = FocusNode();

  void _onFocusChange() {
    if (myFocusNode.hasFocus) {
      _openSearchPage();
      myFocusNode.unfocus();
    }
  }
```

**Comment by andytwoods** [comment]
I guess this could be used: https://stackoverflow.com/questions/44490622/disable-a-text-edit-field-in-flutter. Would be a shame to create a new package just with this feature!

**Comment by jifalops** [comment]
Focus is currently how the input knows to show the picker dialog. I may be able to wrap it in a gesture detector. It may actually make some of the logic cleaner.

**Comment by andytwoods** [comment]
I think you can just do node.unfocus() immediately before launching the date input  function.

**Comment by noordawod** [comment]
Added support for this in #8.

**Comment by jifalops** [comment]
Fixed in v0.1.5
